libcxx/include/__iterator/aliasing_iterator.h

```diff
@@ -102,7 +102,7 @@ struct __aliasing_iterator_wrapper {
 
     _LIBCPP_HIDE_FROM_ABI _Alias operator*() const _NOEXCEPT {
       _Alias __val;
-      __builtin_memcpy(&__val, std::__to_address(__base_), sizeof(value_type));
+      __builtin_memcpy(&__val, reinterpret_cast<void*>(std::__to_address(__base_)), sizeof(value_type));
```

Comment:
IMO this is a false-positive. This inspects the non-trivial object and implicitly starts the lifetime of `_Alias`, which is perfectly defined behaviour.

Reply:
The cast should probably be `static_cast<void*>` also

Reply:
ack for the false positive. I'll move that to a static_cast which should silent the warning.